src/spikeinterface/core/waveform_extractor.py

```diff
@@ -517,15 +524,19 @@ def is_extension(self, extension_name) -> bool:
         if self.folder is None:
             return extension_name in self._loaded_extensions
         else:
-            if self.format == "binary":
-                return (self.folder / extension_name).is_dir() and (
-                    self.folder / extension_name / "params.json"
-                ).is_file()
-            elif self.format == "zarr":
-                return (
-                    extension_name in self._waveforms_root.keys()
-                    and "params" in self._waveforms_root[extension_name].attrs.keys()
-                )
+            # Extensions already loaded in memory
```

Comment:
I also think that this part can benefit from early return after if (if condition: return None) and avoiding the else to decrease nesting.

Reply:
done